plugin-server/src/worker/tasks.ts

```diff
@@ -31,11 +39,40 @@ export const workerTasks: Record<string, TaskRunner> = {
         return hub.pluginSchedule !== null
     },
     reloadPlugins: async (hub) => {
-        // Jitter the reload time to avoid all workers reloading at the same time.
-        const jitterMs = Math.random() * hub.RELOAD_PLUGIN_JITTER_MAX_MS
-        status.info('💤', `Sleeping for ${jitterMs}ms to jitter reloadPlugins`)
-        await sleep(jitterMs)
-        await retryIfRetriable(async () => await setupPlugins(hub))
+        if (RELOAD_PLUGINS_PROMISE && !RELOAD_PLUGINS_PROMISE_STARTED) {
+            // A reload is already scheduled and hasn't started yet. When it starts it will load the
+            // state of plugins after this reload request was issued, so we're done here.
+            return
+        }
+
+        if (RELOAD_PLUGINS_PROMISE && RELOAD_PLUGINS_PROMISE_STARTED) {
+            // A reload was in progress, we need to wait for it to finish and then we can schedule a
+            // new one (or a concurrent request will beat us to it after also waiting here, which is
+            // fine!).
+            await RELOAD_PLUGINS_PROMISE
+        }
+
+        if (!RELOAD_PLUGINS_PROMISE) {
+            // No reload is in progress, schedule one. If multiple concurrent requests got in line
+            // above, we only need one to schedule the reload here.
+
+            RELOAD_PLUGINS_PROMISE = (async () => {
+                // Jitter the reload time to avoid all workers reloading at the same time.
+                const jitterMs = Math.random() * hub.RELOAD_PLUGIN_JITTER_MAX_MS
+                status.info('💤', `Sleeping for ${jitterMs}ms to jitter reloadPlugins`)
+                await sleep(jitterMs)
+
+                RELOAD_PLUGINS_PROMISE_STARTED = true
+                try {
+                    await retryIfRetriable(async () => await setupPlugins(hub))
```

Comment:
Let's set a longer wait here, maybe 5 seconds? No need to re-jitter as the initial try has jitter.